### PR TITLE
Update diffusers integration guide

### PR DESCRIPTION
**Pull request by soumik12345**
## Description

Updates the Huggingface diffusers guide to demonstrate the features included in https://github.com/wandb/wandb/pull/6804

https://github.com/wandb/docodile/assets/19887676/3d4e36ea-d325-41d1-99a8-d41c6dbe1086

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [x] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [ ] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [x] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

File: docs/guides/integrations/diffusers.md [hunk 1]
```diff
@@ -129,8 +175,248 @@ The arguments passed to the `autolog` function is just `init` which accepts a di
 |:--:| 
 | **An example of how the autologger logs the configs of your experiment.** |
 
+## Tracking Multi-pipeline Workflows
+
+This section demonstrates how the Weights & Biases autologger can be used for tracking workflows that involve multiple diffusion pipeline calls. We initially generate an image using the [`AmusedPipeline`](https://huggingface.co/docs/diffusers/api/pipelines/amused) and then use [`StableDiffusionImg2ImgPipeline`](https://huggingface.co/docs/diffusers/api/pipelines/stable_diffusion/img2img) to modify the generated image.
```
Comment (katjacksonWB) [inline docs/guides/integrations/diffusers.md]:
```suggestion
This section demonstrates how the Weights & Biases AutoLogger can be used for tracking workflows that involve multiple diffusion pipeline calls. We initially generate an image using the [`AmusedPipeline`](https://huggingface.co/docs/diffusers/api/pipelines/amused) and then use [`StableDiffusionImg2ImgPipeline`](https://huggingface.co/docs/diffusers/api/pipelines/stable_diffusion/img2img) to modify the generated image.
```

File: docs/guides/integrations/diffusers.md [hunk 2]
```diff
@@ -129,8 +175,248 @@ The arguments passed to the `autolog` function is just `init` which accepts a di
...
+).images
+
+# Finish the experiment
+wandb.finish()
+```
+
+  </TabItem>
+</Tabs>
+
+| ![An example of how the autologger tracks multiple diffusion pipeline calls in a single experiment](@site/static/images/integrations/diffusers-autolog-5.png) |
```
Comment (katjacksonWB) [inline docs/guides/integrations/diffusers.md]:
```suggestion
| ![An example of how the AutoLogger tracks multiple diffusion pipeline calls in a single experiment](@site/static/images/integrations/diffusers-autolog-5.png) | 
```

File: docs/guides/integrations/diffusers.md [hunk 3]
```diff
@@ -129,8 +175,248 @@ The arguments passed to the `autolog` function is just `init` which accepts a di
...
+# Finish the experiment
+wandb.finish()
+```
+
+  </TabItem>
+</Tabs>
+
+| ![An example of how the autologger tracks multiple diffusion pipeline calls in a single experiment](@site/static/images/integrations/diffusers-autolog-5.png) | 
+|:--:| 
+| **An example of how the autologger tracks multiple diffusion pipeline calls in a single experiment.** |
```
Comment (katjacksonWB) [inline docs/guides/integrations/diffusers.md]:
```suggestion
| **An example of how the AutoLogger tracks multiple diffusion pipeline calls in a single experiment.** |
```

**Review by morganmcg1 — COMMENTED** [review]
## (Updated) Clearer intro to using the logger

Update: lets put the description of the autologger arguments into its own header so its super clear

```
## Diffusers Autologger Arguments
The logger accepts.......
```

--- 
Immediately under the header "Next level logging..." can you add a brief section like this that covers:
1) we use an autologger to log to W&B, you just have to import and initialise it
2) The arguments are the same as wandb.init
3) Here are the 2 lines of code to begin logging:

```
# import the autolog function
from wandb.integration.diffusers import autolog

# call the autologger before calling the pipeline
autolog(init=dict(project="diffusers_logging"))
```

And here is a brief end-to-end example of the autologger in action:
(existing code block)
```
import torch
from diffusers import DiffusionPipeline

# import the autolog function
from wandb.integration.diffusers import autolog

# call the autologger before calling the pipeline
autolog(init=dict(project="diffusers_logging"))

# Initialize the diffusion pipeline
pipeline = DiffusionPipeline.from_pretrained(
    "stabilityai/stable-diffusion-2-1", torch_dtype=torch.float16
).to("cuda")

# Define the prompts, negative prompts, and seed.
prompt = [
    "a photograph of an astronaut riding a horse",
    "a photograph of a dragon"
]
negative_prompt = ["ugly, deformed", "ugly, deformed"]
generator = torch.Generator(device="cpu").manual_seed(10)

# call the pipeline to generate the images
images = pipeline(
    prompt,
    negative_prompt=negative_prompt,
    num_images_per_prompt=2,
    generator=generator,
)
```


And then remove this hint, unless you think its valuable to keep as a reminder
Screenshot 2024-02-13 at 10 36 18

File: docs/guides/integrations/diffusers.md [hunk 4]
```diff
@@ -109,8 +149,14 @@ images = pipeline(
...
 )
+
+# Finish the experiment
+wandb.finish()
 ```
 
+  </TabItem>
+</Tabs>
+
 Calling the `autolog` creates a [W&B run](https://docs.wandb.ai/guides/runs). All subsequent pipeline calls are therefore registered and their inputs and outputs are logged to the run. The prompts, negative prompts, and the generated media are logged in a [`wandb.Table`](https://docs.wandb.ai/guides/tables) while all other configs associated with the experiment including seed and the pipeline architecture are stored in the config section for the run. The generated media are also logged in [media panels](https://docs.wandb.ai/guides/track/log/media) in the run.
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: Move this "Calling the `autolog` creates..." section higher up, maybe under the "Next level logging..." section. Its an important concept to understand what its doing

File: docs/guides/integrations/diffusers.md [hunk 5]
```diff
@@ -82,11 +82,51 @@ pip install --upgrade diffusers transformers accelerate wandb
...
+    {label: 'Notebook', value: 'notebook'},
+  ]}>
+  <TabItem value="script">
+
 ```python
 import torch
 from diffusers import DiffusionPipeline
 from wandb.integration.diffusers import autolog
 
+# call the autologger before calling the pipeline
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: Is this code example needed if there is already one above? feels duplicative?

File: docs/guides/integrations/diffusers.md [hunk 6]
```diff
@@ -129,8 +175,248 @@ The arguments passed to the `autolog` function is just `init` which accepts a di
...
+
+| ![An example of how the autologger tracks multiple diffusion pipeline calls in a single experiment](@site/static/images/integrations/diffusers-autolog-5.png) | 
+|:--:| 
+| **An example of how the autologger tracks multiple diffusion pipeline calls in a single experiment.** |
+
+:::info
+When `autolog()` is called, it initializes a Weights & Biases run, which automatically tracks the inputs and the outputs from [all supported pipeline calls](https://github.com/wandb/wandb/blob/main/wandb/integration/diffusers/autologger.py#L12-L72). Each pipeline call is tracked into its own table in the workspace, and the configs associated with the pipeline call is appended to the list of workflows in the configs for that run.
+:::
+
+Let's demonstrate the autologger with a typical [Stable Diffusion XL + Refiner](https://huggingface.co/docs/diffusers/using-diffusers/sdxl#base-to-refiner-model) workflow, in which the latents generated by the [`StableDiffusionXLPipeline`](https://huggingface.co/docs/diffusers/api/pipelines/stable_diffusion/stable_diffusion_xl) is refined by the corresponding refiner.
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: Is this section needed if you've already demonstrated multi-pipeline above?

Comment (morganmcg1) [reply docs/guides/integrations/diffusers.md]: > Try to avoid using first-person plural like 'Let's'."

address this comment from `vale` too

File: docs/guides/integrations/diffusers.md [hunk 7]
```diff
@@ -129,8 +175,248 @@ The arguments passed to the `autolog` function is just `init` which accepts a di
...
+
+  </TabItem>
+</Tabs>
+
+| ![An example of how the autologger tracks multiple diffusion pipeline calls in a single experiment](@site/static/images/integrations/diffusers-autolog-5.png) | 
+|:--:| 
+| **An example of how the autologger tracks multiple diffusion pipeline calls in a single experiment.** |
+
+:::info
+When `autolog()` is called, it initializes a Weights & Biases run, which automatically tracks the inputs and the outputs from [all supported pipeline calls](https://github.com/wandb/wandb/blob/main/wandb/integration/diffusers/autologger.py#L12-L72). Each pipeline call is tracked into its own table in the workspace, and the configs associated with the pipeline call is appended to the list of workflows in the configs for that run.
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: address `vale` comments

File: docs/guides/integrations/diffusers.md (hunk 2 above)
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: @katjacksonWB , @ngrayluna can you add `autologger` to the `vale` whitelist please?

File: docs/guides/integrations/diffusers.md (hunk 1 above)
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: address `vale` comments

**Comment by morganmcg1** [comment]
Swap these 2 code blocks, so that the quick 2-line one comes first

```
## Next-level logging in just 2 lines
Log all the prompts, negative prompts, generated media, and configs associated with your experiment by simply including 2 lines of code:
```
Screenshot 2024-02-13 at 13 24 56


```
## How the Autolog works?

...

Here is a brief end-to-end example of the autologger in action:
```

Screenshot 2024-02-13 at 13 25 01

File: docs/guides/integrations/diffusers.md [hunk 8]
```diff
@@ -78,59 +78,184 @@ pip install --upgrade diffusers transformers accelerate wandb
   </TabItem>
 </Tabs>
 
-## Weights & Biases Autologger for Diffusers
+### How the Autolog works?
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: fix `vale` comment

maybe say "autologger" vs "autolog" ? your call, it should just be consistent throughout

File: docs/guides/integrations/diffusers.md [hunk 9]
```diff
@@ -78,59 +78,184 @@ pip install --upgrade diffusers transformers accelerate wandb
...
 
-| ![An example of how the autologger logs the configs of your experiment](@site/static/images/integrations/diffusers-autolog-3.gif) | 
+| ![An example of how the autolog tracks an SDXL + Refiner experiment](@site/static/images/integrations/diffusers-autolog-6.gif) | 
 |:--:| 
-| **An example of how the autologger logs the configs of your experiment.** |
+| **An example of how the autolog tracks an SDXL + Refiner experiment.** |
 
 :::info
-You can find a list of supported pipeline calls [here](https://github.com/wandb/wandb/blob/main/wandb/integration/diffusers/autologger.py#L12-L67). In case, you want to request a new feature of this integration or report a bug associated with it, please open an issue on [https://github.com/wandb/wandb/issues](https://github.com/wandb/wandb/issues).
+You can find a list of supported pipeline calls [here](https://github.com/wandb/wandb/blob/main/wandb/integration/diffusers/autolog.py#L12-L72). In case, you want to request a new feature of this integration or report a bug associated with it, please open an issue on [https://github.com/wandb/wandb/issues](https://github.com/wandb/wandb/issues).
```
Comment (morganmcg1) [inline docs/guides/integrations/diffusers.md]: actually maybe also bring this up under the "How autolog works" section

**Review by morganmcg1 — APPROVED** [review]

File: docs/guides/integrations/diffusers.md (hunk 8 above)
Comment (soumik12345) [reply docs/guides/integrations/diffusers.md]: I would recommend using `autolog` since its consistent with how we use it in the code.

File: docs/guides/integrations/diffusers.md [hunk 10]
```diff
@@ -10,51 +10,27 @@ import { CTAButtons } from '@site/src/components/CTAButtons/CTAButtons.tsx';
 
 <CTAButtons colabLink="https://colab.research.google.com/github/wandb/examples/blob/master/colabs/diffusers/lcm-diffusers.ipynb"></CTAButtons>
 
-[🤗 Diffusers](https://huggingface.co/docs/diffusers) is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. The W&B integration adds rich, flexible experiment tracking, media visualization, pipeline architecture, and configuration managaement to interactive centralized dashboards without compromising that ease of use.
+[🤗 Diffusers](https://huggingface.co/docs/diffusers) is the go-to library for state-of-the-art pre-trained diffusion models for generating images, audio, and even 3D structures of molecules. The W&B integration adds rich, flexible experiment tracking, media visualization, pipeline architecture, and configuration management to interactive centralized dashboards without compromising that ease of use.
 
 ## Next-level logging in just 2 lines
```
Comment (ngrayluna) [inline docs/guides/integrations/diffusers.md]: Spell out "2" in "Next-level logging in just 2 lines"

**Review by katjacksonWB — APPROVED** [review]
Will add autolog to the Vale allowlist, but otherwise, looks good.
